src/http/mod.rs

```diff
@@ -156,11 +172,22 @@ impl HttpClient {
                 },
                 Err(e) => {
                     metrics::DOCUMENT.with_label_values(&["fetch_error"]).inc();
-                    error!(
-                        "Unable to fetch document, id={}, reason={}, url={}",
-                        req_id, e, url
-                    );
-                    Err(Errors::FetchFailed)
+                    let err = e
+                        .source()
+                        .unwrap()
+                        .downcast_ref::<std::io::Error>()
+                        .unwrap();
+                   
+                    if let ErrorKind::TimedOut = err.kind() {
+                        error!("Unable to fetch document, connection/response/request to the server timed out, id={}", req_id);
+                        Err(Errors::TimedOut)
+                    } else {
+                        error!(
+                            "Unable to fetch document, id={}, reason={}, url={}",
+                            req_id, e, url
+                        );
+                        Err(Errors::FetchFailed)
+                    }
```

Comment:
Trying to think how a consumer of the api would handle the `TimeOut` error as opposed to the catch all `FetchFailed`. Please can you elaborate on what use case did you have in mind?

Reply:
Hyper-timeout is propagating a s std Error to Hyper , that hyper can not understand ... e.g Hyper has a is_timeout () function in its error module, but its fails to detect timeouts by hyper-timeout , the only way I could find to detect timeout was converting the hyper error to its source of std Error and matching against timeout , none of hypers status codes for timeout worked

Reply:
I see, that is good to know. The question was more of whether we should distinguish between a timeout and any other error type and why.

Reply:
before an error was only emitting a fetch failed , but since hyper-timeout forces hyper into an error ... and that is the only place I could find to catch 'timeout has occurred', I matched it to catch just one type by if let , and if it doesn't it just goes back to its default behavior it was emitting before i changed anything , as the errors we want to catch increase we can just extend them above by additional if-let's ... there might be instances where the unwrap actually fails and it can;t downcast_ref... that is probably a bug I have to address , maybe using andthen there

Reply:
I think it would be useful for a consumer to be able to distinguish between a timeout and generic fetch failed (i.e. for ui messaging)